### PR TITLE
fix(builder): vendor library chunks include sources

### PR DESCRIPTION
**Pull request by Asuka109**
## Description

用户项目内路径可能命中拆包测试正则，例如下面的预设：

https://github.com/modern-js-dev/modern.js/blob/e120a8d2368d452d5f027bca4cbb11b99f3691c7/packages/builder/builder/src/plugins/splitChunks.ts#L75-L79

会导致 `./src/pages/history/index.module.less` 被打包到 `lib-router.css` 里，并影响样式加载顺序和优先级。

## Types of changes

- [ ] Docs change / Dependency upgrade
- [x] Bug fix
- [ ] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/builder/builder/src/plugins/splitChunks.ts [hunk 1]
```diff
@@ -58,6 +58,10 @@ function getUserDefinedCacheGroups(forceSplitting: Array<RegExp>): CacheGroup {
   return cacheGroups;
 }
 
+/** Expect to match path just like "./node_modules/react-router/" */
+export const createDependenciesRegExp = (...dependencies: string[]) =>
+  new RegExp(`[\\\\/]node_modules[\\\\/](${dependencies.join('|')})[\\\\/]`);
```
[the remark this answers is not in the document]
Comment (Asuka109) [reply packages/builder/builder/src/plugins/splitChunks.ts:63]: Node.js 和常见的包管理都强依赖 `node_modules/vendor` 这样的目录结构来执行 resolve，各种形式依赖结构的直接和间接依赖也都能符合这种结果，我觉得这样写法应该没啥问题呀

File: .changeset/bright-adults-admire.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@modern-js/module-tools': minor
```
Comment (chenjiahan) [inline .changeset/bright-adults-admire.md:2]: 包名错咯

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
Conflicts need to be solved.

**Comment by Asuka109** [comment]
> Conflicts need to be solved.

done.

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**